File: programmers/[고득점Kit][정렬]/K번째수.py
# 프로그래머스 고득점 Kit K번째 수

def solution(array, commands):
    answer = []

    for num in commands:
        newarray = array[num[0] - 1:num[1]]
        newarray.sort()
        answer.append(newarray[num[2] - 1])

    return answer

# 한줄 풀이
# def solution(array, commands):
#     return list(map(lambda x:sorted(array[x[0]-1:x[1]])[x[2]-1], commands))

File: Algorithm-Swift/Algorithm-Swift/Programmers/문자열/K번째수.swift
//
//  K번째수.swift
//  Algorithm-Swift
//
//  Created by 이원석 on 2021/04/19.
//


import Foundation

func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    return commands.map({(key)in 
            return array[(key[0]-1)...(key[1]-1)].sorted()[key[2]-1]                  
    })
    
}